common/src/main/java/io/netty/util/internal/logging/LocationAwareSlf4JLogger.java

```diff
@@ -54,28 +54,28 @@ public boolean isTraceEnabled() {
     @Override
     public void trace(String msg) {
         if (isTraceEnabled()) {
-            log(TRACE_INT, msg, null);
+            log(TRACE_INT, msg);
         }
     }
 
     @Override
     public void trace(String format, Object arg) {
         if (isTraceEnabled()) {
-            log(TRACE_INT, format, arg);
+            log(TRACE_INT, org.slf4j.helpers.MessageFormatter.format(format, arg).getMessage(), arg);
```

Comment:
Presumably there's no need to also pass the args now? If so it would be better to pass null for the last arg to avoid varargs array allocation.

Reply:
from other implementations I think you still should pass the args.

Reply:
I still don't follow (sorry I'm probably missing something) ... the additional args are only passed for the purpose of later substitution into the message, but now the substitution is done upfront. So the message passed won't have `{}` corresponding to the args..?

Reply:
@njhill when looking at other implementation they all forwarded the "args", so I did the same as I thought maybe they are used somehow later on. I think at worse you have one more alloc but not passing these down may produce other side-effects depending on the underlying implementations.

That said maybe you are right and we should just not do it.

Reply:
@njhill PTAL again... I think this addresses your concern and is also cleaner :)

Reply:
@normanmaurer LGTM, I was about to suggest something similar :)